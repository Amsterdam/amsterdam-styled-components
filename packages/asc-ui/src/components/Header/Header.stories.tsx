import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import styled from '@datapunt/asc-core'
import { ReactComponent as ChevronRight } from '@datapunt/asc-assets/lib/Icons/ChevronRight.svg'
import { ReactComponent as ChevronDown } from '@datapunt/asc-assets/lib/Icons/ChevronDown.svg'
import { ReactComponent as ExternalLink } from '@datapunt/asc-assets/lib/Icons/ExternalLink.svg'
import { ReactComponent as MenuIcon } from '@datapunt/asc-assets/lib/Icons/Menu.svg'
import BoxWrapper from '../../internals/Box/BoxWrapper'
import Header, {
  HeaderWrapperStyle,
  SearchBarStyle,
  MenuBarStyle,
  MenuDropDownStyle,
} from './Header'
import ContentFiller from '../../internals/ContentFiller/ContentFiller'
import Typography from '../Typography'
import SearchBar from '../SearchBar'
import Menu from '../Menu/Menu'
import { MenuItem, MenuLabel } from '../..'
import SubMenu from '../Menu/SubMenu'
import { breakpoint } from '../../utils'

const outsideBackgoundColor = '#E6E6E6'
const contentBackgrountColor = '#ffffff'

const HeaderShortStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall={false}
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth
    />
    <ContentFiller backgroundColor={contentBackgrountColor} />
  </BoxWrapper>
)

const HeaderShortContentStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall={false}
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth={false}
    />
    <ContentFiller backgroundColor={contentBackgrountColor} maxWidth="1800px" />
  </BoxWrapper>
)

const HeaderTallStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth={false}
    />
    <ContentFiller
      backgroundColor={contentBackgrountColor}
      maxWidth="1800px"
      headerSize="tall"
    />
  </BoxWrapper>
)

const HeaderTallWithContentStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth={false}
    >
      <Typography element="p">
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet debitis,
        dicta incidunt maxime necessitatibus voluptatibus.
      </Typography>
    </Header>
    <ContentFiller
      backgroundColor={contentBackgrountColor}
      maxWidth="1800px"
      headerSize="tall"
    />
  </BoxWrapper>
)

const DataportaalHeaderWrapperStyle = styled(HeaderWrapperStyle)`
  ${SearchBarStyle} {
    flex-grow: 1;
    & > div {
      flex-grow: 0;
      width: 100%;
      max-width: 430px;

      @media screen and ${breakpoint('max-width', 'mobileM')} {
        max-width: 245px;
      }
    }
  }

  ${MenuBarStyle} {
    @media screen and ${breakpoint('max-width', 'tablet')} {
      display: none;
    }
  }

  ${MenuDropDownStyle} {
    @media screen and ${breakpoint('min-width', 'tablet')} {
      display: none;
    }
  }
`

const MenuDesktop = (
  <Menu>
    <MenuItem onClick={action('click')}>One</MenuItem>
    <SubMenu label="Two">
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuLabel>Two</MenuLabel>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuItem icon={<ExternalLink />} onClick={action('click')}>
        Show more
      </MenuItem>
    </SubMenu>
    <MenuItem onClick={action('click')}>Three</MenuItem>
  </Menu>
)

const MenuMobile = (
  <Menu icon={<MenuIcon />} mobile>
    <MenuItem onClick={action('click')}>One</MenuItem>
    <SubMenu arrowIcon={<ChevronDown />} label="Two">
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuLabel>Two</MenuLabel>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuItem icon={<ExternalLink />} onClick={action('click')}>
        Show more
      </MenuItem>
    </SubMenu>
    <MenuItem onClick={action('click')}>Three</MenuItem>
  </Menu>
)

const SearchBarDesktop: React.FC<{}> = () => {
  const [searchText, setText] = React.useState('')
  return (
    <SearchBar
      placeholder="Enter the search text"
      onTextChanged={(text: string) => {
        setText(text)
        action(`text changed: ${searchText}`)
      }}
      onSearch={() => {
        action(`button clicked: ${searchText}`)
      }}
      text={searchText}
    />
  )
}

// const SearchBarMobile = (
//   <SearchBar
//     placeholder="Enter the search text"
//     onTextChanged={(text: string) => {
//       setText(text)
//       action(`text changed: ${searchText}`)
//     }}
//     onSearch={() => {
//       action(`button clicked: ${searchText}`)
//     }}
//     text={searchText}
//   />
// )

const HeaderTallDataportaalStory: React.FC<{}> = () => {
  return (
    <BoxWrapper backgroundColor={outsideBackgoundColor}>
      <Header
        tall
        title="Data en informatie"
        homeLink="http://data.amsterdam.nl"
        fullWidth={false}
        styledComponent={DataportaalHeaderWrapperStyle}
      >
        <SearchBarDesktop />
        {MenuDesktop}
        {MenuMobile}
      </Header>
      <ContentFiller
        backgroundColor={contentBackgrountColor}
        maxWidth="1800px"
        headerSize="tall"
      />
    </BoxWrapper>
  )
}

storiesOf('Composed/Header', module)
  .add('Header short', () => <HeaderShortContentStory />)
  .add('Header short, full width', () => <HeaderShortStory />)
  .add('Header tall', () => <HeaderTallStory />)
  .add('Header tall with header content', () => <HeaderTallWithContentStory />)
  .add('Header tall for dataportaal', () => <HeaderTallDataportaalStory />)
