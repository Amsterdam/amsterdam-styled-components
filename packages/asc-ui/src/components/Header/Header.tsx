import React from 'react'
import HeaderStyle, { Props as HeaderStyleProps } from './HeaderStyle'
import HeaderWrapperStyle, {
  Props as HeaderWrapperProps,
} from './HeaderWrapperStyle'
import HeaderLogoText, { LogoProps } from './HeaderLogoText'
import HeaderNavigation from './HeaderNavigation'
import HeaderLinks from './HeaderLinks'
import { CustomCssPropsType } from '../../utils'
import AmsterdamLogo from '../AmsterdamLogo'

type Props = {
  tall?: boolean
  homeLink: string
  title?: string
  navigation?: React.ReactNode
  links?: React.ReactNode
  logo?: React.FC<LogoProps>
} & HeaderWrapperProps &
  HeaderStyleProps &
  CustomCssPropsType &
  React.HTMLAttributes<HTMLDivElement>

const Header: React.FC<Props> = ({
  css,
  title,
  homeLink,
  fullWidth,
  tall,
  navigation,
  links,
  logo,
  ...otherProps
}) => (
  <HeaderWrapperStyle {...{ css, tall }} id="header">
    <HeaderStyle {...{ fullWidth, ...otherProps }}>
      <HeaderLogoText {...{ tall, title, homeLink, logo }} />
      <HeaderNavigation>{navigation}</HeaderNavigation>
      <HeaderLinks>{links}</HeaderLinks>
    </HeaderStyle>
  </HeaderWrapperStyle>
)

Header.defaultProps = {
  tall: false,
  fullWidth: true,
  logo: AmsterdamLogo,
}

export default Header
