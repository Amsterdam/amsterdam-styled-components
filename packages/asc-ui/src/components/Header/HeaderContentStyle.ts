import styled from '@datapunt/asc-core'
import { breakpoint } from '../../utils'

const HeaderContentStyle = styled.div`
  height: 50px;
  margin: 0;
  padding: 15px 25px 0 0;
  font-weight: 700;
  width: 70%;
  display: flex;
  justify-content: normal;
  align-self: flex-start;
  justify-content: flex-end;
  align-items: center;
  flex-shrink: 0;
  box-sizing: border-box;

  @media screen and ${breakpoint('max-width', 'laptopM')} {
    display: none;
  }
`
export default HeaderContentStyle
