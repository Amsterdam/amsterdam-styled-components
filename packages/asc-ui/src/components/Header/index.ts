import HeaderStyle from './HeaderStyle'
import HeaderLinksStyle from './HeaderLinksStyle'
import HeaderTitleStyle from './HeaderTitleStyle'
import HeaderLogoTextStyle from './HeaderLogoTextStyle'
import HeaderNavigationStyle from './HeaderNavigationStyle'

export { default } from './Header'

export const HeaderStyles = {
  HeaderLinksStyle,
  HeaderLogoTextStyle,
  HeaderNavigationStyle,
  HeaderTitleStyle,
  HeaderStyle,
}
