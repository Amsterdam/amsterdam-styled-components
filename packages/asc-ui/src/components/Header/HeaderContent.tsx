import { FunctionComponent, HTMLAttributes, ReactNode } from 'react'
import HeaderContentStyle from './HeaderContentStyle'

type HeaderContentProps = {
  children: ReactNode
} & HTMLAttributes<HTMLDivElement>

const HeaderContent: FunctionComponent<HeaderContentProps> = ({
  children,
  ...otherProps
}) => <HeaderContentStyle {...otherProps}>{children}</HeaderContentStyle>

export default HeaderContent
