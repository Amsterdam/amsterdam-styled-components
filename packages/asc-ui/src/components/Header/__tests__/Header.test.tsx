import * as React from 'react'
import Header from '../Header'
import { renderWithTheme } from '../../../utils/withTheme'
import Typography from '../../Typography'
import 'jest-styled-components'

const setup = (props = {}) =>
  renderWithTheme(
    <Header
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      {...props}
    >
      <Typography>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet debitis,
        dicta incidunt maxime necessitatibus voluptatibus.
      </Typography>
    </Header>,
  )

describe('Header', () => {
  it('should render the short version', () => {
    const component = setup()
    expect(component).toMatchSnapshot()
  })

  it('should render the short version without the full width', () => {
    const component = setup({ fullWidth: false })
    expect(component).toMatchSnapshot()
  })

  it('should render the tall version', () => {
    const component = setup({ tall: true })
    expect(component).toMatchSnapshot()
  })

  it('should render the tall version without a title', () => {
    const component = setup({ tall: true, title: null })
    expect(component).toMatchSnapshot()
  })
})

describe('Header short with content', () => {})
