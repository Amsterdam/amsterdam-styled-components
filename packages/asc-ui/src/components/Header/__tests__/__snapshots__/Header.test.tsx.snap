// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the short version 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 15px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  max-width: none;
}

.c1:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c7 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c7:hover {
  color: #ec0000;
}

.c7:hover {
  color: inherit;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c5 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("-react-component: svg-mock;");
}

.c4 {
  display: inline-block;
  height: 30px;
  width: 68px;
}

.c4:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c2 {
  height: 50px;
  margin: 0;
  font-family: 'AvenirNextLTW01-Demi',sans-serif;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 .c3 {
  margin-right: 10px;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: none;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  position: fixed;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c0 .c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c0 .c6 {
  margin-top: -2px;
}

.c10 {
  line-height: 1.25;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1.25;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media screen and (max-width:375px) {
  .c7 {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c7 {
    font-size: 16px;
  }
}

@media screen and (min-width:1200.02px) {
  .c2 .c3 {
    margin-right: 40px;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <h1
      class="c2"
    >
      <a
        class="c3 c4"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c5"
        />
      </a>
      <a
        class="c6 c7"
        href="http://data.amsterdam.nl"
      >
        Data en informatie
      </a>
    </h1>
    <div
      class="c8 c9"
    >
      <p
        class="c10"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet debitis, dicta incidunt maxime necessitatibus voluptatibus.
      </p>
    </div>
  </header>
</div>
`;

exports[`Header should render the short version without the full width 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 15px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  max-width: 1570px;
}

.c1:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c7 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c7:hover {
  color: #ec0000;
}

.c7:hover {
  color: inherit;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c5 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("-react-component: svg-mock;");
}

.c4 {
  display: inline-block;
  height: 30px;
  width: 68px;
}

.c4:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c2 {
  height: 50px;
  margin: 0;
  font-family: 'AvenirNextLTW01-Demi',sans-serif;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 .c3 {
  margin-right: 10px;
}

.c11 .c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c11 .c6 {
  margin-top: -2px;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: 1800px;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  position: fixed;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c0 .c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c0 .c6 {
  margin-top: -2px;
}

.c10 {
  line-height: 1.25;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1.25;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media screen and (max-width:375px) {
  .c7 {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c7 {
    font-size: 16px;
  }
}

@media screen and (min-width:1200.02px) {
  .c2 .c3 {
    margin-right: 40px;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <h1
      class="c2"
    >
      <a
        class="c3 c4"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c5"
        />
      </a>
      <a
        class="c6 c7"
        href="http://data.amsterdam.nl"
      >
        Data en informatie
      </a>
    </h1>
    <div
      class="c8 c9"
    >
      <p
        class="c10"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet debitis, dicta incidunt maxime necessitatibus voluptatibus.
      </p>
    </div>
  </header>
</div>
`;

exports[`Header should render the tall version 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 15px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  max-width: none;
}

.c3:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c11 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c11:hover {
  color: #ec0000;
}

.c11:hover {
  color: inherit;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c9 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("-react-component: svg-mock;");
}

.c7 {
  display: inline-block;
  height: 30px;
  width: 68px;
}

.c7:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c5 {
  height: 50px;
  margin: 0;
  font-family: 'AvenirNextLTW01-Demi',sans-serif;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 .c6 {
  margin-right: 10px;
}

.c15 .c12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c15 .c10 {
  margin-top: -2px;
}

.c16 .c12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c16 .c10 {
  margin-top: -2px;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: none;
  box-shadow: 0 4px rgba(0,0,0,0.04);
}

.c0 .c12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c14 {
  line-height: 1.25;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1.25;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media screen and (max-width:375px) {
  .c11 {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c11 {
    font-size: 16px;
  }
}

@media screen and (min-width:1200.02px) {
  .c7 {
    height: 68px;
    width: 100px;
  }

  .c7 .c8 {
    background-image: url("-react-component: svg-mock;");
  }
}

@media screen and (min-width:1200.02px) {
  .c5 .c6 {
    margin-right: 40px;
  }
}

@media screen and (max-width:1200px) {
  .c0 {
    position: fixed;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }

  .c0 .c10 {
    margin-top: -2px;
  }
}

@media screen and (min-width:1200.02px) {
  .c0 {
    box-shadow: none;
    position: relative;
  }

  .c0:after {
    content: '';
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    height: 50px;
    margin-top: -50px;
    background-color: #F5F5F5;
  }

  .c0 .c2 {
    z-index: 100;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c0 .c1 {
    background-color: transparent;
  }

  .c0 .c10 {
    margin-top: 0px;
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
    line-height: 38px;
    font-size: 30px;
  }

  .c0 .c4 {
    height: 78px;
    padding: 15px 0;
  }

  .c0 .c12 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    margin-left: 25px;
    margin-right: 25px;
    height: 50px;
  }
}

<div
  class="c0"
>
  <header
    class="c1 c2 c3"
  >
    <h1
      class="c4 c5"
    >
      <a
        class="c6 c7"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c8 c9"
        />
      </a>
      <a
        class="c10 c11"
        href="http://data.amsterdam.nl"
      >
        Data en informatie
      </a>
    </h1>
    <div
      class="c12 c13"
    >
      <p
        class="c14"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet debitis, dicta incidunt maxime necessitatibus voluptatibus.
      </p>
    </div>
  </header>
</div>
`;

exports[`Header should render the tall version without a title 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 15px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  max-width: none;
}

.c3:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c9 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("-react-component: svg-mock;");
}

.c7 {
  display: inline-block;
  height: 30px;
  width: 68px;
}

.c7:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c5 {
  height: 50px;
  margin: 0;
  font-family: 'AvenirNextLTW01-Demi',sans-serif;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 .c6 {
  margin-right: 10px;
}

.c13 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c14 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: none;
  box-shadow: 0 4px rgba(0,0,0,0.04);
}

.c0 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c12 {
  line-height: 1.25;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1.25;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media screen and (min-width:1200.02px) {
  .c7 {
    height: 68px;
    width: 100px;
  }

  .c7 .c8 {
    background-image: url("-react-component: svg-mock;");
  }
}

@media screen and (min-width:1200.02px) {
  .c5 .c6 {
    margin-right: 40px;
  }
}

@media screen and (max-width:1200px) {
  .c0 {
    position: fixed;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }

  .c0 .sc-bxivhb {
    margin-top: -2px;
  }
}

@media screen and (min-width:1200.02px) {
  .c0 {
    box-shadow: none;
    position: relative;
  }

  .c0:after {
    content: '';
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    height: 50px;
    margin-top: -50px;
    background-color: #F5F5F5;
  }

  .c0 .c2 {
    z-index: 100;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c0 .c1 {
    background-color: transparent;
  }

  .c0 .sc-bxivhb {
    margin-top: 0px;
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
    line-height: 38px;
    font-size: 30px;
  }

  .c0 .c4 {
    height: 78px;
    padding: 15px 0;
  }

  .c0 .c10 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    margin-left: 25px;
    margin-right: 25px;
    height: 50px;
  }
}

<div
  class="c0"
>
  <header
    class="c1 c2 c3"
  >
    <h1
      class="c4 c5"
    >
      <a
        class="c6 c7"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c8 c9"
        />
      </a>
    </h1>
    <div
      class="c10 c11"
    >
      <p
        class="c12"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet debitis, dicta incidunt maxime necessitatibus voluptatibus.
      </p>
    </div>
  </header>
</div>
`;
